### ISSUE TITLE
USWDS - Bug: Date picker - mouse hover preventing keyboard interaction on high zoom magnification

### ISSUE THREAD

**Issue by amycole501** [uswds/uswds#5752]
### Describe the bug

When testing the date picker for zoom magnification functionality, I found that at 400% and using a keyboard to select a date I was unable to use the arrow keys to move off a date. I was "stuck" on a date because my mouse was called over to that date automatically.

In essence, the mouse was jumping to focus on a date and wouldn't allow the keyboard to take control of functionality.

I was using Chrome on a PC.

### Steps to reproduce the bug

1. Go to https://designsystem.digital.gov/components/date-picker/
2. Zoom into 400% magnification
3. Try tabbing into the date picker calendar icon; tab into a date to select
4. Then try using keyboard arrow keys to move around inside the calendar. Inconsistently it works; but sometimes if the mouse is anywhere inside the browser it'll "call" the cursor over to the date and create a keyboard trap.

### Expected Behavior

Tabbing into the calendar icon and navigating using a keyboard only should never trap the keyboard focus on a date even at a high magnification like 400%

> [!CAUTION]
> **This is a AA level failure**
> The intent of this Success Criterion is to support people with low vision who need to enlarge text and read it in a single column. When the browser zoom is used to scale content to 400%, it reflows - i.e., it is presented in one column so that scrolling in more than one direction is not necessary. [^1]

[^1]: [WCAG Reflow (Level AA) requirement](https://arc.net/l/quote/qblbjcul)

### System setup

PC, Chrome

### Additional context

This type of keyboard trap inside the calendar is an issue because some low vision users use a high level of magnification and a keyboard to navigate inside components.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds-site/issues) for duplicate bug reports.

**Comment by mahoneycm** [comment]
## Mouseover capture

I was able to replicate this issue! 

It appears to happen whenever the viewport has to move in order the keep the calendar in view. The cursor will automatically snap to the selection (the cursor may be hidden, but moving it slightly will display the mouse hovering over the selected date). When this happens, it prevents keyboard interactions from moving the selection. The mouseover appears to take priority. In my opinion, the mouseover should never prevent the keyboard interaction.

I haven't conducted any formal testing yet but it might have something to do with the mouseover functions:

https://github.com/uswds/uswds/blob/202448875393d39d663614d2d8f9a12c91fed75e/packages/usa-date-picker/src/index.js#L2228-L2240

**Comment by brunerae** [comment]
@amycole501 which WCAG level is this failing? Thank you!

**Comment by amycole501** [comment]
Reflow https://www.w3.org/WAI/WCAG21/Understanding/reflow.html 
However it seems to also happen at lower zoom magnifications but reflow is the one where I found it.

**Comment by mahoneycm** [comment]
Looks like it's a WCAG AA requirement

> Using the responsive web design approach is the most effective method of achieving the goal of allowing people to zoom in to 400%. [^1]
[^1]: (https://www.w3.org/WAI/WCAG21/Understanding/reflow.html#:~:text=Using%20the%20responsive%20web%20design%20approach%20is%20the%20most%20effective%20method%20of%20achieving%20the%20goal%20of%20allowing%20people%20to%20zoom%20in%20to%20400%25.)

**Comment by mejiaj** [comment]
@mahoneycm estimate on complexity?

**Comment by brunerae** [comment]
@mahoneycm / @amyleadem / @mejiaj moving this out of the Sprint cycle since its stale.

**Comment by mahoneycm** [comment]
@brunerae Sorry, I actually have this one in progress still! Fell behind but picked it back up recently and have made great progress. Pulled back in because I'm hoping to have this review ready this week 👍
